File: services/academic-service/src/test/java/com/visor/school/academic/model/AcademicStandingTest.java
package com.visor.school.academic.model;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class AcademicStandingTest {

    @Test
    void shouldHaveAllRequiredAcademicStandingValues() {
        AcademicStanding[] standings = AcademicStanding.values();

        assertTrue(contains(standings, AcademicStanding.GOOD_STANDING));
        assertTrue(contains(standings, AcademicStanding.PROBATION));
        assertTrue(contains(standings, AcademicStanding.SUSPENDED));
        assertTrue(contains(standings, AcademicStanding.GRADUATED));
        assertEquals(4, standings.length);
    }

    @Test
    void shouldHaveCorrectStandingNames() {
        assertEquals("GOOD_STANDING", AcademicStanding.GOOD_STANDING.name());
        assertEquals("PROBATION", AcademicStanding.PROBATION.name());
        assertEquals("SUSPENDED", AcademicStanding.SUSPENDED.name());
        assertEquals("GRADUATED", AcademicStanding.GRADUATED.name());
    }

    private boolean contains(AcademicStanding[] array, AcademicStanding value) {
        for (AcademicStanding standing : array) {
            if (standing == value) {
                return true;
            }
        }
        return false;
    }
}

File: services/academic-service/src/test/kotlin/com/visor/school/academicservice/model/AcademicStandingTest.kt
package com.visor.school.academicservice.model

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class AcademicStandingTest {

    @Test
    fun `should have all required academic standing values`() {
        val standings = AcademicStanding.values()

        assertTrue(standings.contains(AcademicStanding.GOOD_STANDING))
        assertTrue(standings.contains(AcademicStanding.PROBATION))
        assertTrue(standings.contains(AcademicStanding.SUSPENDED))
        assertTrue(standings.contains(AcademicStanding.GRADUATED))
        assertEquals(4, standings.size)
    }

    @Test
    fun `should have correct standing names`() {
        assertEquals("GOOD_STANDING", AcademicStanding.GOOD_STANDING.name)
        assertEquals("PROBATION", AcademicStanding.PROBATION.name)
        assertEquals("SUSPENDED", AcademicStanding.SUSPENDED.name)
        assertEquals("GRADUATED", AcademicStanding.GRADUATED.name)
    }
}

